src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -612,99 +669,159 @@ def realm_rm_zone_creds(self, realm_token_b64):
-
         period = RGWPeriod(period_info)
-
         logging.info('Period: ' + period.id)
-
         zonegroup = period.find_zonegroup_by_name(zonegroup_name)
         if not zonegroup:
-            raise RGWAMException('zonegroup %s not found' % (zonegroup or '<none>'))
+            raise RGWAMException(f'zonegroup {zonegroup_name} not found')
 
         zg = EntityName(zonegroup.name)
-
-        try:
-            zone_info = self.zone_op().create(realm, zg, EntityName(zone_name), endpoints, False,
-                                              access_key, secret)
-        except RGWAMException as e:
-            raise RGWAMException('failed to create zone', e)
-
-        zone_name = zone_info['name']
-        zone_id = zone_info['id']
-
         zone = EntityName(zone_name)
-
-        success_message = 'Created zone %s (%s)' % (zone_name, zone_id)
+        success_message = f'Modified zone {realm_name} {zonegroup_name} {zone_name}'
         logging.info(success_message)
+        try:
+            self.zone_op().modify(zone, zg, endpoints=','.join(endpoints),
+                                  access_key=access_key, secret=secret)
+        except RGWAMException as e:
+            raise RGWAMException('failed to modify zone', e)
 
+        # done, let's update the period
         try:
             period_info = self.period_op().update(realm, zg, zone, True)
         except RGWAMException as e:
             raise RGWAMException('failed to update period', e)
 
         period = RGWPeriod(period_info)
-
         logging.debug(period.to_json())
 
-        svc_id = realm_name + '.' + zone_name
+        return (0, success_message, '')
+
+    def get_realms_info(self):
+        realms_info = []
+        for realm_name in self.realm_op().list():
+            realm = self.get_realm(realm_name)
+            master_zone_inf = self.period_op().get_master_zone(realm)
+            zone_ep = self.period_op().get_master_zone_ep(realm)
+            if master_zone_inf and 'system_key' in master_zone_inf:
+                access_key = master_zone_inf['system_key']['access_key']
+                secret = master_zone_inf['system_key']['secret_key']
+            else:
+                access_key = ''
+                secret = ''
+            realms_info.append({"realm_name": realm_name,
+                                "realm_id": realm.id,
+                                "is_primary": False,
+                                "master_zone_id": master_zone_inf['id'] if master_zone_inf else '',
+                                "endpoint": zone_ep[0] if zone_ep else None,
+                                "access_key": access_key,
+                                "secret": secret})
+        return realms_info
+
+    def zonegroup_create(self, realm_token_b64, zonegroup_name=None,
+                         endpoints=None, zonegroup_is_master=True):
+        if not realm_token_b64:
+            print('missing realm access config')
```

Comment:
was going to comment on the use of `print` here but I see it is used elsewhere in this file as well. Does it actually work properly? Typically in mgr module stuff we always have to log, raise an exception or return things.

Reply:
fixed